packages/dds/map/src/directory.ts

```diff
@@ -1132,22 +1209,29 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
 		}
 
 		// Create the sub directory locally first.
-		const isNew = this.createSubDirectoryCore(subdirName, true);
-
-		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-		const subDir: IDirectory = this._subdirectories.get(subdirName)!;
+		const isNew = this.createSubDirectoryCore(
+			subdirName,
+			true,
+			-1,
+			this.runtime.clientId ?? "detached",
```

Comment:
Is there anything that prevents "detached" from being a legitimate clientId? It's probably not worth protecting against, but it would be unfortunate to ever hit that bug.

Reply:
These are the clientids that we get from push. They have always been unique and we don't expect them to be like "detached".